### PR TITLE
05 interfaces

### PR DESCRIPTION
**Pull request by Happybolt**
Homework solution 05 interfaces.
Implement converter string uppercase using procfs and lowercase converter using sysfs.

### CODE REVIEW

File: 05-Interfaces/string_converter.c [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+	if (pfile == NULL) {
+		pr_err("string_converter: error creating procfs file\n");
+		remove_proc_entry(PROC_DIR, NULL);
+		return -ENOMEM;
+	}
+
+	stfile = proc_create(PROC_STAT, 0666, pdir, &stat_file);
+	if (stfile == NULL) {
+		pr_err("string_converter: error creating procfs file\n");
+		remove_proc_entry(PROC_DIR, NULL);
```
Comment (an1kh) [inline 05-Interfaces/string_converter.c]: pfile won't be removed

File: 05-Interfaces/string_converter.c [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+static struct statistic pstat;
+
+static void conv_uppercase(char *str, size_t count)
+{
+	int i;
+
+	for (i = 0; i < count; i++) {
+		if (str[i] >= 97 && str[i] <= 122) {
+			str[i] -= 32;
```
Comment (an1kh) [inline 05-Interfaces/string_converter.c]: Please do not use magic numbers

File: 05-Interfaces/string_converter.c [hunk 3]
```diff
@@ -35,6 +43,18 @@ static void conv_uppercase(char *str, size_t count)
 
 }
 
+static void conv_lowercase(char *str, size_t count)
+{
+	int i;
+
+	for (i = 1; i < count; i++) {
+		if (str[i] >= 'A' && str[i] <= 'Z') {
+			str[i] += 32;
```
Comment (an1kh) [inline 05-Interfaces/string_converter.c]: what is 32?

**Review by an1kh — CHANGES_REQUESTED** [review]
Please fix inline comments

File: 05-Interfaces/string_converter.c (hunk 1 above)
Comment (Happybolt) [reply 05-Interfaces/string_converter.c]: Fixed, replaced by  "proc_remove(pdir);"

File: 05-Interfaces/string_converter.c (hunk 2 above)
Comment (Happybolt) [reply 05-Interfaces/string_converter.c]: OK, used ASCII

File: 05-Interfaces/string_converter.c (hunk 3 above)
Comment (Happybolt) [reply 05-Interfaces/string_converter.c]: This is a magic number, this number is the result of an arithmetic operation " 'A' - 'a' ". Fixed, used ASCII.

**Review by an1kh — APPROVED** [review]
Well done! Please merge

**Pull request merged**